packages/github-actions/teams-toolkit-cicd/src/operations.ts

```diff
@@ -0,0 +1,102 @@
+import * as path from 'path'
+import {Execute} from './utils/exec'
+import {Commands, Pathes, Miscs, ActionOutputs} from './constant'
+import {ProgrammingLanguage} from './enums/programmingLanguages'
+import * as fs from 'fs-extra'
+import * as core from '@actions/core'
+import {LanguageError, SpfxZippedPackageMissingError} from './errors'
+import {BuildMapQuerier} from './buildMapQuerier'
+
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class Operations {
+  static async BuildTeamsApp(
+    projectRoot: string,
+    capabilities: string[]
+  ): Promise<void> {
+    // Get the project's programming language from env.default.json.
+    const envDefaultPath = path.join(projectRoot, Pathes.EnvDefaultJson)
```

Comment:
Nit: To avoid to miss projectRoot, I suggest that you can make the Pathes.EnvDefaultJson a function that must take a parameter.

Reply:
Good idea. Got it.